### PR TITLE
refactor: Convert auth.js to => .ts

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Converted `auth.js` to `.ts`

### CODE REVIEW

File: src/cli/commands/auth/index.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const apiUrl = url.parse(config.API);
+const authUrl = apiUrl.protocol + '//' + apiUrl.host;
+const debug = Debug('snyk-auth');
+let attemptsLeft = 0;
+
+function resetAttempts() {
+  attemptsLeft = 30;
+}
+
+function webAuth(via) {
```
Comment (kyegupov) [inline src/cli/commands/auth/index.ts]: it'd be nice to label it async (to help against calling it accidentally as sync), even without conversion to await

File: src/cli/commands/auth/index.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  })
+    // clear spinnger in case of success or failure
+    .then(spinner.clear(lbl))
+    .catch((error) => {
+      spinner.clear(lbl)();
+      throw error;
+    });
+}
+
+function testAuthComplete(token) {
```
Comment (kyegupov) [inline src/cli/commands/auth/index.ts]: likewise, async

File: src/cli/commands/auth/index.ts (hunk 1 above)
Comment (kyegupov) [reply src/cli/commands/auth/index.ts]: ...and add a type to mysterious "via"

File: src/cli/commands/auth/index.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          return resolve(testAuthComplete(token));
+        }
+
+        reject(TokenExpiredError());
+      }, 1000);
+    });
+  });
+}
+
+function auth(api, via) {
```
Comment (kyegupov) [inline src/cli/commands/auth/index.ts]: ditto, async

File: src/cli/commands/auth/is-authed.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import * as snyk from '../../../lib';
+import * as config from '../../../lib/config';
+import * as request from '../../../lib/request';
```
Comment (kyegupov) [inline src/cli/commands/auth/is-authed.ts]: do we have an async/await version of that lib/request?

File: src/cli/commands/index.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ require('../../lib/spinner').isRequired = false;
 // time as low as possible
 
 var commands = {
-  auth: hotload('./auth'),
+  auth: (hotload('./auth')),
```
Comment (kyegupov) [inline src/cli/commands/index.js]: does this make a difference?

File: src/lib/errors/custom-error.ts [hunk 6]
```diff
@@ -1,8 +1,9 @@
 export class CustomError extends Error {
```
Comment (kyegupov) [inline src/lib/errors/custom-error.ts]: "CustomError" is not a particularly helpful name, I'd like to find out what's so special about it...

File: src/lib/is-ci.ts [hunk 7]
```diff
@@ -16,6 +16,8 @@ var ciEnvs = [
   'TF_BUILD',
 ];
 
-module.exports = !!Object.keys(process.env).filter(function (env) {
-  return ciEnvs.indexOf(env) !== -1;
-}).length;
+export function isCI(): boolean {
+  return !!Object.keys(process.env)
```
Comment (kyegupov) [inline src/lib/is-ci.ts]: nit: 
const ciEnvs = new Set([...]);
...
return Object.keys(process.env).some((key) => ciEnvs.has(key));

File: src/cli/commands/auth/index.ts (hunk 3 above)
Comment (kyegupov) [reply src/cli/commands/auth/index.ts]: and api should be apiToken

File: src/cli/commands/index.js (hunk 5 above)
Comment (lili2311) [reply src/cli/commands/index.js]: nope left over, removing

File: src/cli/commands/auth/is-authed.ts (hunk 4 above)
Comment (lili2311) [reply src/cli/commands/auth/is-authed.ts]: not yet 😉

File: src/lib/errors/custom-error.ts (hunk 6 above)
Comment (lili2311) [reply src/lib/errors/custom-error.ts]: we use this across different codebases as a default `CustomError`, this is our go to before customising further

**Pull request merged**